### PR TITLE
build(build): resolve workspace version and support BUILD_VERSION override (#1527)

### PR DESCRIPTION
**Pull request by lzjqsdd**
# Summary

Resolve the Cargo workspace version during packaging and allow builds to override the artifact version with `BUILD_VERSION`. Release builds now derive `BUILD_VERSION` from the pushed tag, keeping packaged artifacts and generated Rust version constants aligned with the release version.

# Issue Describe / Design

Related to #1527.

This PR implements the first version-management step:

- The previous `grep '^version ='` did not match the indented `[workspace.package] version`, so `build/build.sh` could resolve the wrong package version.
- `BUILD_VERSION` is now the explicit release/CI override for packaging and generated version constants.
- Release workflow derives `BUILD_VERSION` from the tag name and strips the leading `v` before `make dist`.
- Missing or unparseable `[workspace.package].version` now fails instead of publishing artifacts with an `unknown` version.
- When `BUILD_VERSION` is not set, behavior falls back to the Cargo workspace version, so existing local builds remain stable.

# Changes

| Module / File | Change | Impact on existing behavior |
| ------------- | ------ | --------------------------- |
| `.github/workflows/release.yml` | Export `BUILD_VERSION` from the release tag before `make dist` | Release artifacts and generated binary version output follow the tag version |
| `build/version.sh` | Parse workspace version, apply non-empty `BUILD_VERSION`, and fail when workspace version cannot be resolved | Invalid workspace metadata now fails the build instead of producing `unknown` artifacts |
| `build/build.sh` | Use `get_build_version` instead of root-level `grep`; document `BUILD_VERSION` in help | Local build version now resolves from `[workspace.package] version` |
| `crates/core/curvine-sys/build.rs` | Rebuild when `BUILD_VERSION` changes; override `PKG_VERSION` and `VERSION` | Generated version constants honor release override |
| `crates/core/curvine-sys/src/version.rs` | Add version constant tests and ignore empty `BUILD_VERSION` in override assertions | None |
| `build/tests/test_workspace_version.sh` | Add workspace version, override, empty override fallback, and missing version tests | None |

# Test verified

| Test case | Result | Notes |
| --------- | ------ | ----- |
| `bash build/tests/test_workspace_version.sh` | PASS | workspace version, `BUILD_VERSION` override, empty override fallback, missing version failure |
| `cargo test -p curvine-sys` | PASS | 14 passed |
| `make format` | PASS | pre-commit and clippy checks clean |
| `git diff --check origin/main...HEAD` | PASS | no whitespace errors |

# Dependencies

- Related PRs: None
- Related issues: #1527
- Blocks / blocked by: None

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -164,7 +168,7 @@ fi
 ARCH_NAME=$(get_arch_name)
```
Comment (szbr00) [inline build/build.sh:168]: The release workflow exports only `RELEASE_VERSION`, while this path and `curvine-sys/build.rs` read `BUILD_VERSION`. Therefore a tagged release can retain `0.2.0` in `build-version` and binary version output while only the outer archive uses the tag. Please derive `BUILD_VERSION` from the release tag, stripping the `v` prefix, before invoking `make dist`.

File: build/version.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+get_build_version() {
+  local cargo_toml="${1:?Cargo.toml path required}"
+  if [ -n "${BUILD_VERSION:-}" ]; then
+    printf '%s\n' "$BUILD_VERSION"
+    return 0
+  fi
+  local version
+  version="$(get_workspace_version "$cargo_toml")"
+  printf '%s\n' "${version:-unknown}"
+}
```
Comment (szbr00) [inline build/version.sh:53]: A missing or unparseable workspace version currently succeeds as `unknown`, allowing incorrectly versioned artifacts to be published. Please return a non-zero status when `[workspace.package].version` cannot be resolved and add a negative test for that case.

File: crates/core/curvine-sys/src/version.rs [hunk 3]
```diff
@@ -14,3 +14,23 @@
...
+        assert!(!VERSION.is_empty());
+    }
+
+    #[test]
+    fn build_version_override_is_honored_when_set() {
+        if let Ok(build_version) = std::env::var("BUILD_VERSION") {
+            assert_eq!(PKG_VERSION, build_version);
+            assert_eq!(VERSION, build_version);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (lzjqsdd) [reply crates/core/curvine-sys/src/version.rs:38]: Thanks. Fixed in 9f5d436c: the test now skips empty BUILD_VERSION values, matching build.rs behavior.

File: build/version.sh (hunk 2 above)
Comment (lzjqsdd) [reply build/version.sh:53]: Thanks. Fixed in 9f5d436c: get_build_version now returns non-zero when workspace package version cannot be resolved, with a negative shell test.

File: build/build.sh (hunk 1 above)
Comment (lzjqsdd) [reply build/build.sh:168]: Thanks. Fixed in 9f5d436c: release.yml now derives BUILD_VERSION from RELEASE_VERSION with the leading v stripped and exports it before make dist.

**Review by szbr00 — APPROVED** [review]
